monero/seed.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+# Electrum - lightweight Bitcoin client
+# Copyright (C) 2011 thomasv@gitorious
+#
+# Permission is hereby granted, free of charge, to any person
+# obtaining a copy of this software and associated documentation files
+# (the "Software"), to deal in the Software without restriction,
+# including without limitation the rights to use, copy, modify, merge,
+# publish, distribute, sublicense, and/or sell copies of the Software,
+# and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+# BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+# Note about US patent no 5892470: Here each word does not represent a given digit.
+# Instead, the digit represented by a word is variable, it depends on the previous word.
+#
+# Copied 17 February 2018 from MoneroPy, originally from Electrum:
+#     https://github.com/bigreddmachine/MoneroPy/blob/master/moneropy/mnemonic.py ch: 80cc16c39b16c55a8d052fbf7fae68644f7a5f02
+#     https://github.com/spesmilo/electrum/blob/master/lib/old_mnemonic.py ch:9a0aa9b4783ea03ea13c6d668e080e0cdf261c5b
+
+from monero import wordlists
+from binascii import crc32, hexlify
+from os import urandom
+
+class Seed(object):
+    """Creates a seed object either from local system randomness or an imported phrase.
+
+    :rtype: :class:`Seed <monero.seed.Seed>`
+    """
+
+    n = 1626
+    wordlist = wordlists.english.wordlist # default english for now
+
+    seed_phrase = "" #13 or 25 word mnemonic word string
+    seed_hex = "" # hexadecimal
+
+    def __init__(self):
+        pass
```

Comment:
It would be great if it accepted phrase or hex in the constructor and reacted accordingly (validate then store). With no parameters it could randomize the seed automatically.

Reply:
Ok, added more logic to the constructor that accepts strings and determines if it's hex, 13 word, 24 word, etc..